include/boost/thread/concurrent_queues/detail/sync_deque_base.hpp

```diff
@@ -55,7 +55,7 @@ namespace detail
     inline void close();
 
     inline underlying_queue_type underlying_queue() {
-      lock_guard<mutex> lk(mtx_);
```

Comment:
While you are right that the cost is minimal, I don't see why to pay for if we can avoid it.